### PR TITLE
feat: improve general styles

### PR DESCRIPTION
**Pull request by lessp**
**Ticket:** [#137](https://trello.com/c/RcKfEfIm/137-improve-overall-styling)
**Intent:**

Improve some low hanging fruit

**How to test (optional):**

### All of the following commands should pass.

- [x] `npm test`
- [x] `npm run lint`
- [ ] `npm run cypress`

### Browser testing

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] IE11
- [ ] Edge

### CODE REVIEW

File: src/components/Flex.tsx [hunk 1]
```diff
@@ -2,6 +2,8 @@ import styled from '@emotion/styled'
 import {
   alignItems,
   AlignItemsProps,
+  alignSelf,
+  AlignSelfProps,
```
Comment (momentiris) [inline src/components/Flex.tsx]:
```suggestion
  AlignSelfProps,
  position,
  PositionProps,
```

File: src/components/Flex.tsx [hunk 2]
```diff
@@ -28,6 +31,7 @@ const Flex = styled.div<FlexContainerProps>`
   display: flex;
 
   ${alignItems}
+  ${alignSelf}
```
Comment (momentiris) [inline src/components/Flex.tsx]:
```suggestion
  ${alignSelf}
  ${position}
```

File: src/components/Layout/RegistrationLayout.tsx [hunk 3]
```diff
@@ -15,16 +16,15 @@ interface RegistrationLayoutProps {
...
   display: flex;
+  justify-content: center;
   align-items: center;
+  flex-direction: column;
+
   & > div:last-child {
-    flex-basis: 80%;
+    flex: 1;
   }
 `
```
Comment (momentiris) [inline src/components/Layout/RegistrationLayout.tsx]:
```suggestion
const ArrowContainer = styled.div`
position: absolute;
left: 0;
`
```

File: src/components/Layout/RegistrationLayout.tsx [hunk 4]
```diff
@@ -51,27 +51,29 @@ const RegistrationLayout: React.FC<
   childFnArgs = {},
 }) => {
   return (
-    <Container className="layout">
+    <Grid gridTemplateRows="auto 1fr auto" height="100vh" p="large">
       <NavigationContainer>
```
Comment (momentiris) [inline src/components/Layout/RegistrationLayout.tsx]:
```suggestion
      <Flex alignItems="center" flexDirection="column" position="relative">
```

File: src/components/Layout/RegistrationLayout.tsx [hunk 5]
```diff
@@ -51,27 +51,29 @@ const RegistrationLayout: React.FC<
...
-        <div
-          onClick={() => history.back()}
-          onKeyUp={() => history.back()}
-          role="button"
-          tabIndex={0}
-        >
-          <img alt="Go back" src={Icon} />
-        </div>
+        <Flex alignSelf="stretch" justifyContent="center" mb="small">
```
Comment (momentiris) [inline src/components/Layout/RegistrationLayout.tsx]:
```suggestion

```

File: src/components/Layout/RegistrationLayout.tsx [hunk 6]
```diff
@@ -51,27 +51,29 @@ const RegistrationLayout: React.FC<
...
+            role="button"
+            tabIndex={0}
+          >
+            <img alt="Go back" src={Icon} />
+          </div>
 
-        <StepIndicator step={step} />
+          <StepIndicator step={step} />
+        </Flex>
+        {headerText}
```
Comment (momentiris) [inline src/components/Layout/RegistrationLayout.tsx]:
```suggestion
        <Paragraph>{headerText}</Paragraph>
```

File: src/components/Typography.tsx [hunk 7]
```diff
@@ -90,3 +98,11 @@ Paragraph.defaultProps = {
   mt: 'small',
   mb: 'medium',
 }
+
+Bold.defaultProps = {
+  fontFamily: 'default',
+  fontSize: 'medium',
+  fontWeight: '700',
```
Comment (momentiris) [inline src/components/Typography.tsx]:
```suggestion
  fontWeight: 700,
```

File: src/components/Flex.tsx (hunk 1 above)
Comment (lessp) [reply src/components/Flex.tsx]: The back-icon is 27px wide, I subtracted the this amount from the step-indicator. If the icon changes, we'll obviously have to change that number too, but should be OK!

**Comment by lessp** [comment]
> Looks nice but left some comments with small suggestions. In RegistrationLayout.tsx, the StepIndicator isnt centered. I tried to outline with suggestions how we could change it, but basically:
> 
> 1. Add positionprops to Flex.tsx
> 2. remove const NavigationContainer  = styled.div... and just use <Flex position="relative" {...restofflexprops}>

Cool, should be OK now!

File: src/components/Typography.tsx (hunk 7 above)
Comment (lessp) [reply src/components/Typography.tsx]: styled-system is actually fine with either!

**Comment by momentiris** [comment]
Aha. I got type error

mån 13 maj 2019 kl. 12:09 skrev Tom Ekander <notifications@github.com>:

File: src/components/Typography.tsx (hunk 7 above)
Comment (lessp) [reply src/components/Typography.tsx]: Aha, maybe not! 😄

File: src/components/Flex.tsx (hunk 1 above)
Comment (momentiris) [reply src/components/Flex.tsx]: awesome!

**Pull request merged**